lib/config.js

```diff
@@ -186,17 +188,21 @@ class Config {
     /**
      * Config options
      * @param {Object} options Options to be passed in
+     * @param {Linter} linterContext Linter instance object
      */
-    constructor(options) {
+    constructor(options, linterContext) {
```

Comment:
Intuitively, it seems like a config shouldn't need to know what `linter` instance created it. It seems like it would be better for configs to not mutate linters, and to just abstractly represent config files.

However, this is definitely an improvement over mutating global state, so I think this is fine for now and we can improve it later.